src/main/java/com/arpnetworking/metrics/impl/TsdLogSink.java

```diff
@@ -0,0 +1,218 @@
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of <code>Sink</code> for the query log file.
+ *
+ * @author Ville Koskela (vkoskela at groupon dot com)
+ */
+public class TsdLogSink extends BaseFileSink {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void record(final Event event) {
+
+        try {
+            getMetricsLogger().info(_objectMapper.writeValueAsString(event));
+        } catch (final IOException e) {
+            _logger.warn("Exception recording event", e);
+        }
+    }
+
+    private static Encoder<ILoggingEvent> createEncoder(final boolean immediateFlush) {
+        final PatternLayoutEncoder encoder = new PatternLayoutEncoder();
+        encoder.setPattern("%msg%n");
+        encoder.setImmediateFlush(immediateFlush);
+        return encoder;
+    }
+
+    /**
+     * Protected constructor.
+     *
+     * @param builder Instance of <code>Builder</code>.
+     */
+    protected TsdLogSink(final Builder builder) {
+        this(builder, OBJECT_MAPPER, LOGGER);
+    }
+
+    // NOTE: Package private for testing
+    /* package private */ TsdLogSink(final Builder builder, final ObjectMapper objectMapper, final org.slf4j.Logger logger) {
+        super(builder, createEncoder(builder._immediateFlush.booleanValue()));
+        _objectMapper = objectMapper;
+        _logger = logger;
+    }
+
+    private final ObjectMapper _objectMapper;
+    private final org.slf4j.Logger _logger;
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(StenoLogSink.class);
+
+    static {
+        final SimpleModule simpleModule = new SimpleModule("TsdLogSink");
+        simpleModule.addSerializer(Event.class, EventSerializer.newInstance());
+        simpleModule.addSerializer(Quantity.class, QuantitySerializer.newInstance());
+        OBJECT_MAPPER.setSerializationInclusion(JsonInclude.Include.ALWAYS);
+        OBJECT_MAPPER.registerModule(simpleModule);
+    }
+
+    private static final class EventSerializer extends JsonSerializer<Event> {
+
+        public static JsonSerializer<Event> newInstance() {
+            return new EventSerializer();
+        }
+
+        @Override
+        public void serialize(
+                final Event event,
+                final JsonGenerator jsonGenerator,
+                final SerializerProvider provider)
+                throws IOException {
+
+            jsonGenerator.writeStartObject();
+            jsonGenerator.writeStringField("version", "2f");
```

Comment:
No. These are independent sinks or formatters. Incidentally they both represent the version tag in the serialization using the same characters. Open to a convincing argument otherwise.